### ISSUE TITLE
Add support for read / writers

### ISSUE THREAD

**Issue by realvictorprm** [tracasa/FME-DotNetSDK#7]
Help, 
Although I'm still getting into the FME SDK API, I think it's right that the .net sdk doesn't support the reader and writer extension functionality yet.

Because I need this for the company I would appreciate if this would be implemented next.

**Comment by realvictorprm** [comment]
Hello @ahuarte47 :smiley: I need help for writing the C++ Wrapper Code, is there anything you can give me to ease the process?

**Comment by ahuarte47** [comment]
Hello @realvictorprm, thanks for your idea. We did not need to implement the reader/writer wrappers for our project. Sorry, I know **this new feature is very interesting** but, unfortunately this project was finished some time ago and We have no founds to add new features like this to this SDK. 

Did you have a look at "$(FME_HOME)\pluginbuilder\samples\cpp\code\FMEReaderWriter" folder? It is a good point to start.

**Comment by realvictorprm** [comment]
Yes I already started looking into it. I'm trying to implement it for the SDK.

**Comment by ahuarte47** [comment]
ok, what help would you like?

**Comment by realvictorprm** [comment]
I try to get the general idea of how interop there works.

Is it right that for testing it's not needed to wrap all methods?

Am 16.08.2017 9:18 AM schrieb "Alvaro Huarte" <notifications@github.com>:

**Comment by ahuarte47** [comment]
You can write a full IFMEReader/IFMEWriter implementation (Empty methods is ok to start) in C++/CLI which call a new interface similar to [this](https://github.com/tracasa/FME-DotNetSDK/blob/master/FMEDotNetWrapperFactory/FMEDotNetFactory.h#L52). Then, in the .NET client side, you can write your implementation with empty methods too. The key of this is to load the .NET assembly in the C++/CLI (Maybe in the C++ constructor) with something similar to [this](https://github.com/tracasa/FME-DotNetSDK/blob/master/FMEDotNetWrapperFactory/FMEDotNetWrapperFactory.cpp#L419).

Sorry, I do not know this is useful for you

**Comment by ahuarte47** [comment]
Fortunately, there are few public methods to define for IFMEReader/IFMEWriter classes.

**Comment by realvictorprm** [comment]
I need to look into the section with the App Domain. Looks like I need to
refresh my knowledge

Am 16.08.2017 9:42 AM schrieb "Alvaro Huarte" <notifications@github.com>:

**Comment by realvictorprm** [comment]
My plan would be to define the *interfaces* in the C++ CLI code and the DotNetWrapper *class*. 

@ahuarte47 So far I understood you there is some kind of assembly loading needed afterwards?

**Comment by ahuarte47** [comment]
Yes, in a point of the C++/CLI code you have to load the DotNetWrapper class deployed in your specific .DLL file. I would write something similar to [this](https://github.com/tracasa/FME-DotNetSDK/blob/master/FMEDotNetWrapperFactory/FMEDotNetWrapperFactory.cpp#L404)
